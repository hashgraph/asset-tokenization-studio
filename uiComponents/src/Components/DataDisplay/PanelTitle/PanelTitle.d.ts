/// <reference types="react" />
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";
import type { TextProps } from "@/Components/Foundations/Text";
import type { FlexProps } from "@chakra-ui/react";
export declare const panelTitlePartsList: Array<"title" | "container">;
type Parts = typeof panelTitlePartsList;
export interface PanelTitleThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
}
export interface PanelTitleProps extends Omit<FlexProps, "children"> {
    title: string;
    titleProps?: TextProps;
}
export declare const PanelTitle: ({ title, titleProps, ...props }: PanelTitleProps) => JSX.Element;
export {};
