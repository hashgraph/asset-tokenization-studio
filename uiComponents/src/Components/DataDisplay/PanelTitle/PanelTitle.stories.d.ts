import type { PanelTitleProps } from "@Components/DataDisplay/PanelTitle";
declare const _default: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, PanelTitleProps>;
export default _default;
export declare const PanelTitleWithBorderBottom: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, PanelTitleProps>;
