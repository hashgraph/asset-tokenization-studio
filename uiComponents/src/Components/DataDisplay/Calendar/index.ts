export * from "./types";
export * from "./Calendar";
export * from "./CalendarPanel";
export * from "./context";
export * from "./helpers";
