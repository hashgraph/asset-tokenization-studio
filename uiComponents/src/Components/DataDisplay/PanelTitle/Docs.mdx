import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as PanelTitleStories from "./PanelTitle.stories";

<Meta of={PanelTitleStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=2091-20098&t=7YJ1vWLNVVuMBa0t-0"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `PanelTitle.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    PanelTitle: ConfigPanelTitle,
  },
};
```

```tsx dark
const ConfigPanelTitle: PanelTitleThemeConfiguration = {
  parts: panelTitlePartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"title" | "container"` (`panelTitlePartsList`)

## Props

<ArgTypes of={PanelTitleStories} />

## Stories

<Stories title="" />
