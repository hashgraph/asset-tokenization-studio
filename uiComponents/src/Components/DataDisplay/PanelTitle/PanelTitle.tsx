import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";
import {
  Flex,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/react";
import React from "react";
import type { SystemStyleObject as ChakraSystemStyleObject } from "@chakra-ui/system";
import { Text } from "@/Components/Foundations/Text";
import type { TextProps } from "@/Components/Foundations/Text";
import type { FlexProps } from "@chakra-ui/react";

export const panelTitlePartsList: Array<"title" | "container"> = [
  "title",
  "container",
];

type Parts = typeof panelTitlePartsList;

export interface PanelTitleThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {}

type PanelTitleThemeStyle = Record<Parts[number], ChakraSystemStyleObject>;

export interface PanelTitleProps extends Omit<FlexProps, "children"> {
  title: string;
  titleProps?: TextProps;
}

export const PanelTitle = ({
  title,
  titleProps,
  ...props
}: PanelTitleProps) => {
  const styles = useChakraMultiStyleConfig(
    "PanelTitle"
  ) as PanelTitleThemeStyle;
  return (
    <Flex sx={styles.container} {...props}>
      <Text sx={styles.title} {...titleProps}>
        {title}
      </Text>
    </Flex>
  );
};
