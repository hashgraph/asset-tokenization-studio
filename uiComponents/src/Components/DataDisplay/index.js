export * from "./Accordion";
export * from "./Calendar";
export * from "./DefinitionList";
export * from "./DetailReview";
export * from "./Dropdown";
export * from "./FileCard";
export * from "./InfoDivider";
export * from "./Table";
export * from "./Tabs";
export * from "./Tag";
export * from "./PanelTitle";
