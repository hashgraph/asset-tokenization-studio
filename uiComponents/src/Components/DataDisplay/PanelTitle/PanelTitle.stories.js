var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { PanelTitle } from "@Components/DataDisplay/PanelTitle";
var props = {
    title: "Panel Title"
};
export default {
    title: "Design System/Data Display/PanelTitle",
    component: PanelTitle,
    args: {},
    parameters: {
        docs: {}
    }
};
var Template = function (args) { return React.createElement(PanelTitle, __assign({}, args)); };
export var PanelTitleWithBorderBottom = Template.bind({});
PanelTitleWithBorderBottom.args = __assign({}, props);
