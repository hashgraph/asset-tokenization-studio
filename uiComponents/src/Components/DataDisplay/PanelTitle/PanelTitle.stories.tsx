import React from "react";
import type { Meta, StoryFn } from "@storybook/react";
import type { PanelTitleProps } from "@Components/DataDisplay/PanelTitle";
import { PanelTitle } from "@Components/DataDisplay/PanelTitle";

const props: PanelTitleProps = {
  title: "Panel Title",
};

export default {
  title: "Design System/Data Display/PanelTitle",
  component: PanelTitle,
  args: {},
  parameters: {
    docs: {},
  },
} as Meta<typeof PanelTitle>;

const Template: StoryFn<typeof PanelTitle> = (args) => <PanelTitle {...args} />;

export const PanelTitleWithBorderBottom = Template.bind({});

PanelTitleWithBorderBottom.args = {
  ...props,
};
