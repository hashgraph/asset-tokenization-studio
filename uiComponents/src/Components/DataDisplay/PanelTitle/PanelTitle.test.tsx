import { render } from "@/test-utils";
import React from "react";
import type { PanelTitleProps } from "@Components/DataDisplay/PanelTitle";
import { PanelTitle } from "@Components/DataDisplay/PanelTitle";

describe("<PanelTitle/>", () => {
  const defaultText = "Panel title";
  const defaultProps: PanelTitleProps = {
    title: defaultText,
  };

  const factoryComponent = (props?: Partial<PanelTitleProps>) =>
    render(<PanelTitle {...defaultProps} {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
    expect(component.getByText(defaultText)).toBeVisible();
  });
});
