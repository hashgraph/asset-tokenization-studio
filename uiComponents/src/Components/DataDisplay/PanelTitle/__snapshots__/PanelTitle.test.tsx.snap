// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PanelTitle/> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-146esfn"
  >
    <p
      class="css-dmlij3"
    >
      Panel title
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
